src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -314,10 +314,13 @@ public async Task<bool> ReadAsync(DbDataReader reader)
             // Parse Results into Json and return
             if (dbDataReader.HasRows)
             {
-                // Make sure to get the complete json string in case of large document.
-                jsonResult =
-                    JsonSerializer.Deserialize<TResult>(
-                        await GetJsonStringFromDbReader(dbDataReader));
+                string jsonString = await GetJsonStringFromDbReader(dbDataReader);
+
+                if (!string.IsNullOrEmpty(jsonString))
```

Comment:
Json string can be null or empty when the returned result from the db is NULL. This is one of the test cases we have. For other databases this case would return 0 rows, because we are not constructing any json in the sql query. in dw case this would return a single row with an empty string owing to the isNull check.

Reply:
So for other dbs, HasRows check returns false right? For those situations, we would log the subsequent message and return a default value for the `jsonResult`. in case of DW, we will miss logging this information and yet return a default value. 

Could you please:
1) Add this reasoning in the code,
2) Log the information even in case of DW..

Reply:
For sure, let me add this.